### PR TITLE
Implement directly ChannelOutboundHandlerAdapter in BookieProtoEncoding#ResponseEncoder

### PR DESCRIPTION
**Pull request by eolivelli**
This change is mostly a clean up/refactor which drops intermediate MessageToMessageEncoder and MessageToMessageDecoder from BookieProtoEncoding

This is a manual cherry pick of #1286

### CODE REVIEW

**Comment by eolivelli** [comment]
All tests are passing on my laptop on Java 8. Automatic QA on CI won't work on branch-4.6

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtoEncoding.java [hunk 1]
```diff
@@ -355,122 +356,127 @@ private static ByteBuf serializeProtobuf(MessageLite msg, ByteBufAllocator alloc
...
+                ReferenceCountUtil.release(msg);
             }
         }
     }
 
-    @Sharable
-    public static class ResponseEncoder extends MessageToMessageEncoder<Object> {
-        final EnDecoder REP_PREV3;
-        final EnDecoder REP_V3;
+    @Sharable
```
Comment (ivankelly) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtoEncoding.java]: why isn't checkstyle complaining about trailing space?

Comment (eolivelli) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtoEncoding.java]: checkstyle is passing, will double check. thank you

**Comment by eolivelli** [comment]
@ivankelly maybe on 4.6 we did not have full checkstyle coverage.
I have stripped all trailing white spaces from that class.

**Comment by ivankelly** [comment]
lgtm. Shipit

**Comment by eolivelli** [comment]
I have passed a patched 4.6.2-SNAPSHOT JAR with this change to my upstream collegues. All the successful tests were with a patched BK 4.7-SNAPSHOT.
The cherry pick was not automatic, maybe on 4.6 the change won't have effect (I was testing with 4.6.1 when I started this saga, now 4.6.2 changed a bit, on BookieProtoEncoding).

I will merge as soon as I have positive feedback.

**Comment by eolivelli** [comment]
merged to branch-4.6

**Pull request closed without merging**